### PR TITLE
Issue 773: As a logged out user, I should not see the header menu with Welcome, Reports, Profile, Admin links

### PR DESCRIPTION
**Pull request by jorgegonzalez**
### Summary of Changes
This PR hides the Nav menu when the user is not logged in.

_Addresses issue #773
_Acceptance criteria as stated in the issue_

### How to Test
- Open the app and ensure you are not logged in.
- Check to ensure you do *not* see the navigation menu with the Welcome, Reports, Profile, Admin links.
- Upon logging in, you should be able to see the navigation menu with its respective links.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] As a logged out user, I should not see the header menu with Welcome, Reports, Profile, Admin links
  - [x] As a logged in user, I want to review the header links and check for any access control issues by user/role type

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are frontend code style checks passing on CircleCI?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces frontend code, is that code documented both inline and overall?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
  - No issues found.

Screen Shot 2021-04-02 at 10 10 57 AM
Screen Shot 2021-04-02 at 10 28 58 AM

### CODE REVIEW

File: tdrs-frontend/src/components/Header/Header.jsx [hunk 1]
```diff
@@ -19,13 +19,10 @@ import NavItem from '../NavItem/NavItem'
...
   const pathname = useSelector((state) => state.router.location.pathname)
   const user = useSelector((state) => state.auth.user)
+  const authenticated = useSelector((state) => state.auth.authenticated)
 
   const isSystemAdmin = () => {
-    if (user && user.roles) {
-      return user.roles.some((role) => role.name === 'System Admin')
-    }
-
-    return false
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Header/Header.jsx:28]: Just cleaning this up as the branching logic was unnecessary

File: tdrs-frontend/src/components/Header/Header.test.js [hunk 2]
```diff
@@ -243,4 +243,24 @@ describe('Header', () => {
...
+      <Provider store={store}>
+        <Header />
+      </Provider>
+    )
+
+    expect(queryByText('Welcome')).not.toBeInTheDocument()
+    expect(queryByText('Data Files')).not.toBeInTheDocument()
+    expect(queryByText('Profile')).not.toBeInTheDocument()
+    expect(queryByText('Admin')).not.toBeInTheDocument()
+  })
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Header/Header.test.js:265]: Test that the nav links do *not* render

File: tdrs-frontend/src/components/Header/Header.jsx [hunk 3]
```diff
@@ -50,23 +47,31 @@ function HeaderComp() {
...
+                    href="/edit-profile"
+                  />
+                  {isSystemAdmin() && (
+                    <NavItem
+                      pathname={pathname}
+                      tabTitle="Admin"
+                      href={`${process.env.REACT_APP_BACKEND_HOST}/admin/`}
+                    />
+                  )}
+                </>
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Header/Header.jsx:74]: Prevent the NavItems from rendering based on the `authenticated` result from above

**Comment by alexsoble** [comment]
@jorgegonzalez You may have already considered this, so feel free to ignore if it's not helpful, but would it make sense to use the `PrivateTemplate.jsx` template for this?

The code comments for `PrivateTemplate.jsx` say: "A layout for all pages that require an authenticated context."

**Review by carltonsmith — APPROVED** [review]
LGTM

**Comment by jorgegonzalez** [comment]
> @jorgegonzalez You may have already considered this, so feel free to ignore if it's not helpful, but would it make sense to use the PrivateTemplate.jsx template for this?

Hmm I see how we could do there but it would require a bit of more rework with the way we have the Header - Alert - and Main Content layout set up in App.js. I don't think it would be helpful enough at this stage to warrant that.

**Comment by alexsoble** [comment]
@jorgegonzalez Sounds good, just wanted to make sure it was on your radar! Thanks!

**Review by ADPennington — APPROVED** [review]
ltgm 👍🏾 

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
	- [x]  As a logged out user, I should not see the header menu with Welcome, Reports, Profile, Admin links
	- [x]  As a logged in user, I want to review the header links and check for any access control issues by user/role type

	*as system admin, i can see welcome, data files, profile, and admin tabs in header*
	*as data prepper, i can see welcome, data files, and profile tabs in header*
	*as OFA admin, i can see welcome, data files, and profile tabs in header*

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?

  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
_it's passing but noticing [here](https://app.codecov.io/gh/raft-tech/TANF-app) that 2% of code isnt tested and seems relaed to logout redirects and authorization checks_  


### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are frontend code style checks passing on CircleCI?


### Deliverable 4: Accessible

_n/a: bug fix_ 

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


+ [x] If this PR introduces frontend code, is that code documented both inline and overall?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_noting here that during testing, I stumbled into unexpected behavior: clicking `log out` from the django admin site doesn't log me out of anything. it says logged out, but i can still get back to the admin site (as long as I've not logged out from tdp-frontend.app.cloud.gov. i'll create a separate issue, as this is beyond the scope of this PR_

**Pull request merged**